integration/vscode/ada/src/extension.ts

```diff
@@ -41,6 +42,45 @@ let alsTaskProvider: vscode.Disposable[] = [
     ),
 ];
 
+/**
+ * Add a subprogram box above the subprogram enclosing the cursor's position, if any.
+ *
+ * @example:
+ *
+ *  -------
+ *  - Foo -
+ *  -------
+ *
+ *  procedure Foo is
+ */
+function addSupbrogramBox() {
+    const activeEditor = vscode.window.activeTextEditor;
+
+    getSubprogramSymbol(activeEditor)
+        .then(symbol => {
+            if (symbol !== null) {
+                const name: string = symbol.name ?? ""
+                const insertPos = new vscode.Position(symbol.range.start.line, 0);
+                const indentationRange = new vscode.Range(insertPos, symbol.range.start)
+                const indentation: string = activeEditor?.document.getText(indentationRange) ?? ""
+
+                // Generate the subprogram box after retrieving the indentation of the line of
+                // the subprogram's body declaration.
+                let text: string = indentation + "---" + '-'.repeat(name.length) + "---\n"
```

Comment:
instead of using "\n" systematically, we should probably query the document for the right end of line character -- looks like there is something here https://code.visualstudio.com/api/references/vscode-api  (search for "eol")

Reply:
Indeed, I will have a look